narwhals/typing.py

```diff
@@ -32,4 +33,9 @@ def join(self, *args: Any, **kwargs: Any) -> Any: ...
 # Anything which can be converted to a Narwhals DataFrame.
 IntoDataFrame: TypeAlias = Union["NativeDataFrame", "DataFrame"]
 
-__all__ = ["IntoExpr", "IntoDataFrame"]
+
+def assert_never(_: NoReturn) -> NoReturn:
+    raise AssertionError("Expected code to be unreachable")
+
+
+__all__ = ["IntoExpr", "IntoDataFrame", "assert_never"]
```

Comment:
not sure `assert_never` should be a public type, it's only used internally here

Reply:
If we put a leading underscore, it will make I private *to the typing module*, it is internal, but there are no naming conventions for internal members. That is, it is used in other modules of the package however not intended to be used outside the package